### PR TITLE
Pass through horizontal and vertical covariances

### PR DESCRIPTION
**Pull request by dgonano**
**Test data / coverage**
Coming.. if desired for this minor change. 

**Describe problem solved by the proposed pull request**
For the vision input to the ekf there is only single variance figure used (posErr in the structs) for position representing a spherical error around said position. Some companion computer sensor systems have different errors on horizontal and vertical positions. I guess this is one of the reasons you can set the height mode independently via the EKF2_HGT_MODE flag.
I came across the issue where I wanted to influence the horizontal position accurately, but influence the vertical position only a little (or varying variances).

**Describe your preferred solution**
I've expanded the struct in ecl and included a separate hgtErr which is used on this line instead. I've modified Firmware to push the variances from the vision mavlink message into `vehicle_local_position_s.eph/epv`. It's all tested and working with the variance correctly influencing the EKF. 

**This PR relies on https://github.com/PX4/ecl/pull/507 to be merged first, then i can update the submodule tracking in a commit here.**

**Additional bugfix**
The original variances for `posErr` were never actually being copied into `vision_position.eph`  in `mavlilnk_reciever.cpp`, so the struct has either 0's or randomly initialized data from memory. End result, mavlink msg variances weren't be used to fuse into the EKF

### CODE REVIEW

File: src/modules/mavlink/mavlink_receiver.cpp [hunk 1]
```diff
@@ -1253,6 +1253,10 @@ MavlinkReceiver::handle_message_vision_position_estimate(mavlink_message_t *msg)
 	vision_position.y = pos.y;
 	vision_position.z = pos.z;
 
+	//copy horizontal and vertical covariances
+	vision_position.eph = fmaxf(pos.covariance[0], pos.covariance[6]);
```
Comment (dagar) [inline src/modules/mavlink/mavlink_receiver.cpp]: Why?

**Comment by dagar** [comment]
Could you undo the eclipse project file move? The actual project file isn't versioned because it will constantly change on different systems with different eclipse and plugin versions.

File: src/modules/mavlink/mavlink_receiver.cpp (hunk 1 above)
Comment (dgonano) [reply src/modules/mavlink/mavlink_receiver.cpp]: what do you mean why?

**Comment by dgonano** [comment]
sure, will do

**Comment by TSC21** [comment]
@dgonano I ask: what's the reason to have this PR, as https://github.com/PX4/Firmware/pull/9895 already addresses this and much more? I don't think it makes sense to duplicate efforts.

File: src/modules/mavlink/mavlink_receiver.cpp (hunk 1 above)
Comment (dagar) [reply src/modules/mavlink/mavlink_receiver.cpp]: Why is eph the max of covariance index 0 and covariance index 6?
Why is epv the covariance at index 11?
Why are these magic numbers not defined anywhere?

Comment (mhkabir) [reply src/modules/mavlink/mavlink_receiver.cpp]: Those are just the indices of the diagonal terms for position, but yeah we should document/define them somewhere.

Comment (dagar) [reply src/modules/mavlink/mavlink_receiver.cpp]: Yes I know, just trying to illustrate why magic numbers generally aren't good thing.

**Comment by dgonano** [comment]
This PR basically includes a bug fix on the horizontal var and small changes to add the vertical. It's not meant to replace or undermine your efforts @TSC21, I just did a PR because the changes were simple, minimal, and useful (and I had already done them to get stuff working).

File: src/modules/ekf2/ekf2_main.cpp [hunk 2]
```diff
@@ -1173,8 +1173,9 @@ void Ekf2::run()
 			ev_data.quat = q;
 
 			// position measurement error from parameters. TODO : use covariances from topic
-			ev_data.posErr = fmaxf(_ev_pos_noise.get(), fmaxf(ev_pos.eph, ev_pos.epv));
+			ev_data.posErr = fmaxf(_ev_pos_noise.get(), ev_pos.eph);
 			ev_data.angErr = _ev_ang_noise.get();
+			ev_data.hgtErr = fmaxf(_ev_pos_noise.get(), ev_pos.epv);
```
Comment (TSC21) [inline src/modules/ekf2/ekf2_main.cpp]: The only change required is this one. The changes on the `mavlink_receiver` are going to be replaced by https://github.com/PX4/Firmware/pull/9895.

Comment (dgonano) [reply src/modules/ekf2/ekf2_main.cpp]: Yeah that's fine, I'm not worried if they get overwritten. I was just thinking it's just a an intermediate step until the ODOM one you are working on is ready :)

**Comment by mhkabir** [comment]
@dgonano looks good!

@dagar random failure on Jenkins.

**Comment by dagar** [comment]
@mhkabir yes, we need to track it down and stop tolerating failures.

**Pull request merged**